File: data_types_project/cpp/unions.cpp
#include <iostream>
using namespace std;

union Number {
    int intValue;
    float floatValue;
};

int main() {
    Number number;

    number.intValue = 25;
    cout << "Integer value: " << number.intValue << endl;

    number.floatValue = 3.14;
    cout << "Float value: " << number.floatValue << endl;

    return 0;
}

File: data_types_project/rust/unions.rs
enum Number {
    IntValue(i32),
    FloatValue(f32),
}

fn main() {
    let value1 = Number::IntValue(25);
    let value2 = Number::FloatValue(3.14);

    match value1 {
        Number::IntValue(num) => println!("Integer value: {}", num),
        Number::FloatValue(num) => println!("Float value: {}", num),
    }

    match value2 {
        Number::IntValue(num) => println!("Integer value: {}", num),
        Number::FloatValue(num) => println!("Float value: {}", num),
    }
}